### PR TITLE
Feature/dx 255 update invoice

### PR DESCRIPTION
**Pull request by marinafrydenb**

### CODE REVIEW

**Comment by marinafrydenb** [comment]
@asbjornu what happened to my beautiful branch?

File: payments/invoice/index.md [hunk 1]
```diff
@@ -23,273 +25,236 @@ sidebar:
...
+        },
+        "metadata": { //<-- Optional
+            "key1": "value1",
+            "key2": 2,
+            "key3": 3.1,
+            "key4": false
+
+        },
+        "billingAddress": {
+                "firstName": "firstname/companyname",
```
Comment (asbjornu) [inline payments/invoice/index.md]: Please use test data for the code examples.

```suggestion
                "firstName": "Leia",
```

File: payments/invoice/redirect.md [hunk 2]
```diff
@@ -19,35 +21,53 @@ sidebar:
...
 
 {:.table .table-striped}
-| *POST Request* | **Sweden** ![Swedish flag][se-png] | **Norway** ![Norwegian flag][no-png] | **FInland** ![Finish flag][fi-png] |
-| *Operation* | FinancingConsumer | FinancingConsumer | FinancingConsumer |
-| *Intent* | Authorization | Authorization | Authorization |
-| *Currency* | SEK | NOK | EUR |
-| *InvoiceType* | PayExFinancingSE | PayExFinancingNO | PayExFinancingFI |
+| | **Sweden** ![Swedish flag][se-png]| **Norway** ![Norwegian flag][no-png] | **Finland** ![Finish flag][fi-png] |
+| `operation` | `FinancingConsumer` | `FinancingConsumer` | `FinancingConsumer` |
+| `intent` | `Authorization` | `Authorization` | `Authorization` |
```
Comment (asbjornu) [inline payments/invoice/redirect.md]: Can't `intent` be removed entirely if it's always set to `Authorization`?

```suggestion
```

File: payments/invoice/redirect.md [hunk 3]
```diff
@@ -56,22 +76,344 @@ sequenceDiagram
...
+      "prices": [
+      {
+        "type": "Invoice",
+        "amount": 1500,
+        "vatAmount": 0
+        }
+      ],
+      "description": "Test Purchase",
+      "payerReference": "SomeReference",
+      "generateRecurrenceToken": "true|false", //<-- Optional (default value is false)
```
Comment (asbjornu) [inline payments/invoice/redirect.md]:
```suggestion
      "generateRecurrenceToken": false,
```

File: payments/invoice/direct.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  postal number from the consumer, and make another `POST` request towards
+  PayEx in order to retrieve the name and address from the consumer.
+* To create the authorization transaction, you need to calculate the final
+  price / amount, and make a third `POST` request where you send in
+  the consumer data.
+* To get the authorization result, you need  to follow up with a `GET`
+  request using the `paymentID` received in the first step.
+* Finally, when you are ready to ship your order, you will have to make
+  a `POST` request to make a Capture. **At this point Swedbank Pay will generate
+  the invoice to the consumer.**
```
Comment (NullableInt) [inline payments/invoice/direct.md]: Creating a invoice statement in five easy steps:

1. Collect all purchase information send it to Swedbank Pay
2. Collect personal information and send it to PayEx
3. Create a authorization transaction
4. Retrieve the authorized payment
5. When the order is ready to be shipped, send a capture request

File: payments/invoice/direct.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  the invoice to the consumer and the order is ready for shipping.**
+
+## Options before posting a payment
+
+All valid options when posting a payment with operation equal to
+`FinancingConsumer`, are described in
+[other features][other-features-financing-consumer].
+
+{:.table .table-striped}
+| | **Sweden** ![Swedish flag][se-png]| **Norway** ![Norwegian flag][no-png] | **Finland** ![Finish flag][fi-png] |
```
Comment (asbjornu) [inline payments/invoice/direct.md]: Turn this into a proper table with a header/body separator line and then format the table. This applies to all tables.

```suggestion
| | Sweden ![Swedish flag][se-png]| Norway ![Norwegian flag][no-png] | Finland ![Finish flag][fi-png] |
| :------ | :------------------- | :------- | :------------------- |
```

File: payments/invoice/direct.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+| `operation` | `FinancingConsumer` | `FinancingConsumer` | `FinancingConsumer` |
+| `intent` | `Authorization` | `Authorization` | `Authorization` |
+| `currency` | SEK | NOK | EUR |
```
Comment (asbjornu) [inline payments/invoice/direct.md]: The currencies are code values.

```suggestion
| `currency` | `SEK` | `NOK` | `EUR` |
```

File: payments/invoice/direct.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+[capture]: /payments/credit-card/after-payment#Capture
+[fi-png]: /assets/img/fi.png
+[financing-invoice-1-png]: /assets/img/checkout/test-purchase.png
+[financing-invoice-2-png]: /assets/screenshots/invoice/redirect-view/iframe-verify-data.png
+[no-png]: /assets/img/no.png
+[se-png]: /assets/img/se.png
+[callback-api]: /payments/invoice/other-features#callback
+[hosted-view]: /payments/#hosted-view-implementation
+[optional-features]: /payments/invoice/optional-features
```
Comment (asbjornu) [inline payments/invoice/direct.md]: What's the difference between "optional features" and "other features"?

File: payments/invoice/redirect.md [hunk 8]
```diff
@@ -56,22 +75,342 @@ sequenceDiagram
...
+{:.code-header}
+**Request**
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "payment": {
+    "operation": "FinancingConsumer",
```
Comment (asbjornu) [inline payments/invoice/redirect.md]: Indent code examples with 4 spaces, please. The `prettyJson` extension can help with that.

File: payments/invoice/redirect.md [hunk 9]
```diff
@@ -19,35 +21,52 @@ sidebar:
...
+`FinancingConsumer`, are described in
+[other features][other-features-financing-consumer].
 
 {:.table .table-striped}
-| *POST Request* | **Sweden** ![Swedish flag][se-png] | **Norway** ![Norwegian flag][no-png] | **FInland** ![Finish flag][fi-png] |
-| *Operation* | FinancingConsumer | FinancingConsumer | FinancingConsumer |
-| *Intent* | Authorization | Authorization | Authorization |
-| *Currency* | SEK | NOK | EUR |
-| *InvoiceType* | PayExFinancingSE | PayExFinancingNO | PayExFinancingFI |
+| | **Sweden** ![Swedish flag][se-png]| **Norway** ![Norwegian flag][no-png] | **Finland** ![Finish flag][fi-png] |
```
Comment (asbjornu) [inline payments/invoice/redirect.md]: Make this into a proper table, please.

File: payments/invoice/redirect.md [hunk 10]
```diff
@@ -56,22 +75,342 @@ sequenceDiagram
...
+
+{:.code-header}
+**Response**
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "payment": {
```
Comment (asbjornu) [inline payments/invoice/redirect.md]: Indent all code examples by 4 spaces, please.

**Pull request merged**